core/data/src/main/java/com/google/samples/apps/nowinandroid/core/data/repository/OfflineFirstNewsRepository.kt

```diff
@@ -44,14 +44,13 @@ class OfflineFirstNewsRepository @Inject constructor(
     private val network: NiaNetworkDataSource,
 ) : NewsRepository {
 
-    override fun getNewsResources(): Flow<List<NewsResource>> =
-        newsResourceDao.getNewsResources()
-            .map { it.map(PopulatedNewsResource::asExternalModel) }
-
     override fun getNewsResources(
-        filterTopicIds: Set<String>,
+        query: NewsResourceQuery,
     ): Flow<List<NewsResource>> = newsResourceDao.getNewsResources(
-        filterTopicIds = filterTopicIds,
+        useFilterTopicIds = query.filterTopicIds != null,
```

Comment:
See suggestion below, this could be changed to just 
```suggestion
        filterTopicIds = query.filterTopicIds,
        filterNewsIds = query.filterNewsIds,
```